File: Number Loop Pattern.py
n = int(input("Enter n:"))
ctr=1
for i in range (0,(2*n-1)):
    l=[n]*(2*n-1)
    mi=(2*n-2)//2
    if (i<=mi):
        l[mi]=n-i
    else:
        ctr+=1
        l[mi]=ctr
    me=l[mi]
    terms=((2*me-2)//2)
    for k in range(1,terms+1):
        l[mi-k]=me
        l[mi+k]=me
    for v in range(1,mi+1):
        if (mi-terms-v > 0):
            l[mi-terms-v]=me+v
            l[mi+terms+v]=me+v
    print(l)

File: Number_Loop_Pattern.cpp
/*
Enter n:4
4 4 4 4 4 4 4 
4 3 3 3 3 3 4 
4 3 2 2 2 3 4 
4 3 2 1 2 3 4 
4 3 2 2 2 3 4 
4 3 3 3 3 3 4 
4 4 4 4 4 4 4 
*/
#include <iostream>
using namespace std;

int main() {
    int n;
    cout << "Input: " << endl;
    cin >> n;
    int a[(2*n)-1][(2*n)-1];
    int center = n-1;
    for (int i=0; i<((2*n)-1); i++) {
        for (int j=0; j<((2*n)-1); j++) {
            a[i][j]=max(abs(i-center),abs(j-center))+1;
            cout << a[i][j];
        }
        cout << "\n";
    }
    return 0;
}
